src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -400,6 +408,9 @@ private static function remove_json_comments( $array ) {
 	 * @return array Custom Post Type for the user's origin config.
 	 */
 	public static function get_user_data_from_wp_global_styles( $theme, $create_post = false, $post_status_filter = array( 'publish' ) ) {
+		if ( ! static::theme_has_support() ) {
```

Comment:
The only edge case I can think of is a theme that uses `theme.json` and removes it in a later version. In that case, this changes the behavior: the code will now ignore user data while it was used before.

Reply:
YES! Love this change.

Reply:
That's a fair point. However, I would consider this indeed an edge-case which I don't think is worth addressing here. If a theme removed its `theme.json` it would come with a whole lot of other behavioral changes anyway - I don't think realistically any theme author is going to change a block theme to become a classic theme.